modules/ting_subsearch/ting_subsearch.module

```diff
@@ -0,0 +1,239 @@
+ */
+function ting_subsearch_opensearch_pre_execute($request) {
+  if (get_class($request) != 'TingClientSearchRequest') {
+    return;
+  }
+
+  // These modules needs the facet genreCategory returned from the well, so
+  // ensure it is set if one or both of them are enabled.
+  if (module_exists('ting_subsearch_translate') || module_exists('ting_subsearch_secondary')) {
+    $facets = $request->getFacets();
+    if (!in_array('facet.genreCategory', $facets)) {
+      $facets[] = 'facet.genreCategory';
+      $request->setFacets($facets);
+    }
+  }
+}
+
+/**
+ * Implements hook_theme().
+ */
+function ting_subsearch_theme() {
+  return [
+    'ting_subsearch_message' => [
+      'variables' => [
+        'message' => NULL,
+        'type' => NULL,
+        'suggestion' => NULL,
+        'suggestion_num_results' => NULL,
+        'original_search' => NULL,
+        'original_search_num_results' => NULL,
+      ]
+    ]
+  ];
+}
+
+/**
+ * Theme preprocess function for ting_subsearch_message.
+ */
+function template_preprocess_ting_subsearch_message(&$variables) {
+  $variables['classes_array'] = [];
+  $variables['attributes_array'] = [];
+  $type = $variables['type'];
+  $variables['theme_hook_suggestions'][] = 'ting_subsearch_message__' . $type;
+  $variables['attributes_array']['id'] = 'ting-subsearch-message-' . $type;
+}
+
+/**
+ * Theme process function for ting subsearch message.
+ */
+function template_process_ting_subsearch_message(&$variables) {
+  $variables['classes'] = implode(' ', $variables['classes_array']);
+  $variables['attributes'] = drupal_attributes($variables['attributes_array']);
+}
+
+/**
+ * Returns default HTML for ting subsearch message.
+ */
+function theme_ting_subsearch_message($variables) {
+  $message = $variables['message'];
+  $classes = $variables['classes'];
+  $attributes = $variables['attributes'];
+
+  $output = '<div class="ting-subsearch-message-wrapper">';
+  $output .= "<div class='ting-subsearch-message $classes'$attributes>";
+  $output .= $message;
+  $output .= '</div></div>';
+
+  return $output;
+}
+
+/**
+ * Do a separate secondary search request.
+ *
+ * @param string $keys
+ *   The search phrase.
+ * @param array $conditions
+ *   The conditions to pass to the ting_search execute hook.
+ *
+ * @return TingSearchResultInterface
+ *   The search result of the secondary search.
+ */
+function ting_subsearch_do_secondary_search($keys, $conditions = []) {
+  // Abuse ting_search but remember current search result and set again after
+  // we're done to avoid side effects.
+  $current_search_result = ting_search_current_results();
+
+  // If a profile is active but no context is set, we set our own to ensure
+  // ting_field_search will activate.
+  if (ting_field_search_get_active_profile() && !ting_field_search_get_context()) {
+    ting_field_search_set_context('ting_subsearch');
+  }
+
+  drupal_static_reset('ting_search_search_execute');
+  search_data($keys, 'ting_search', $conditions);
```

Comment:
We're experimenting with some facet support on the secondary search, and ting_search has some handling of facets which we could use. So I'd like to keep this way for now.

It also sets default sort which could be important for secondary search.

Yes it has some processing we don't need, but I think it's acceptable.

Reply:
Switch to a new method of executing subsearchs using the original `$search_result` object with access to user search request via `$search_result->getSearchRequest()`.

This way we only need to add our suggestions keys to the search request and the search will be executed with same filters and sort as ting_search. Only ting_field_search profile needs some custom handling, since its query is added in `opensearch_pre_execute()` and is not present on the original search request object.